x-pack/metricbeat/module/googlecloud/pubsub/_meta/data_subscription.json

```diff
@@ -0,0 +1,53 @@
+{
+    "@timestamp": "2020-01-14T11:46:00.000Z",
+    "@metadata": {
+        "beat": "metricbeat",
+        "type": "_doc",
+        "version": "8.0.0"
+    },
+    "googlecloud": {
+        "labels": {
```

Comment:
just curious is the label key here resource.subscription_id? I understand the value is `test-subscription`, but not sure whats the `resource` part. Thanks!

Reply:
This is more like a property but we thought it was a bit too "complex" to talk about metadata, labels, resources, properties... so we joined all of them into "labels".

This information comes on each [response](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries) of the Stackdriver API. The one you mention here is a resource label as you can see in the link (follow [MonitoredResource](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.uptimeCheckConfigs#MonitoredResource) link too). You can also find "system" labels, "user" labels and instead of having a bunch of `user.labels.*`, `resource.labels.*`, `system.labels.*` we swapped the `.labels` part to just have everything under the key `labels.*`

For example, resource labels in `compute` are the instance type or the storage type. User labels are a key-value you can set in gcp console.

In this case, PubSub don't have the specific implementation to gather `user` labels so we are missing the data that the user might input in the console. This data uses PubSub API and not Stackdriver API, that's why each metricset requires an specific implementation inside the stackdriver metricset

I'm not sure if I did a very good job explaining this now, frankly :sweat_smile: